src/charm.py

```diff
@@ -29,11 +35,15 @@ def __init__(self, msg: str, status_type=None):
 
 
 class Operator(CharmBase):
+    _stored = StoredState()
+
     def __init__(self, *args):
         super().__init__(*args)
 
         self.log = logging.getLogger(__name__)
         self.image = OCIImageResource(self, "oci-image")
+        self._stored.set_default(hash_salt=_gen_pass())
+        self._stored.set_default(side_bar_tabs=Path("src/config.json").read_text())
```

Comment:
While storedstate seems like a reasonable way of doing things, there are some key downsides with it in charms. It is unit specific and it will be reset if the unit restarts. How is this going to affect this feature? Can we avoid using stored state?

Reply:
I think stored-state for kubernetes applications is by default stored on the controller, which I think makes it resistant to pod restarts.  I haven't tried it though